src/table/table-item.class.ts

```diff
@@ -106,6 +106,11 @@ export class TableItem {
 	 */
 	colSpan = 1;
 
+	/**
+	 * The original index of data
+	 */
+	index?: number;
+
```

Comment:
We shouldn't apply an index to each table item as at that point, it becomes redundant. Ideally, we should store the index in `row`. 🤔